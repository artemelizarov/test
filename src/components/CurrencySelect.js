import React from "react";

function CurrencySelect(props) {
  const { label, onChange, currencies, name } = props;
  const currencyCode = props.currency;

  return (
    <div className="CurrencySelect">
      <label htmlFor={name}>{label}</label>
      <select name={name} value={currencyCode} onChange={onChange}>
        {currencies.map(currency => (
          <option key={currency.code} value={currency.code}>
            {currency.code + " - " + currency.name}
          </option>
        ))}
      </select>
    </div>
  );
}

export default CurrencySelect;
