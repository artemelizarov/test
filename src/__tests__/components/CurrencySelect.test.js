import CurrencySelect from "../../components/CurrencySelect";

describe("render CurrencySelect component", () => {
  let wrapper, _onChange;

  beforeAll(() => {
    _onChange = jest.fn();

    wrapper = mount(
      <CurrencySelect
        label="Base currency"
        name="baseCurrency"
        currencies={[
          { code: "USD", name: "United States dollar" },
          { code: "EUR", name: "Euro" }
        ]}
        currency="USD"
        onChange={event => {
          _onChange(event.target.value);
        }}
      />
    );

    let input;
    input = wrapper.find("select").first();
    input.instance().value = "EUR";
    input.simulate("change");
  });

  test("render CurrencySelect component", () => {
    expect(wrapper.find(".CurrencySelect").length).toBe(1);
  });

  test("_onChange is called", () => {
    expect(_onChange).toBeCalledWith("EUR");
  });
});

describe("snapshot-test CurrencySelect component", () => {
  test("Renders correct properties", () => {
    shallowExpect(
      <CurrencySelect
        label="Base currency"
        name="baseCurrency"
        currencies={[
          { code: "USD", name: "United States dollar" },
          { code: "EUR", name: "Euro" }
        ]}
        currency="USD"
        onChange={jest.fn()}
      />
    ).toMatchSnapshot();
  });
});
