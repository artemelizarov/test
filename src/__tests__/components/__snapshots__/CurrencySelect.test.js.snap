// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot-test CurrencySelect component Renders correct properties 1`] = `
<div
  className="CurrencySelect"
>
  <label
    htmlFor="baseCurrency"
  >
    Base currency
  </label>
  <select
    name="baseCurrency"
    onChange={[MockFunction]}
    value="USD"
  >
    <option
      key="USD"
      value="USD"
    >
      USD - United States dollar
    </option>
    <option
      key="EUR"
      value="EUR"
    >
      EUR - Euro
    </option>
  </select>
</div>
`;
